engine.py

```diff
@@ -102,13 +108,19 @@ def host_info(self):
     def execute_command(self, cmd_key):
         """
         Executes a given command.
+
+        Note: This is part of a legacy pathway.
         """
         cb = self.commands[cmd_key]["callback"]
-        
-        if not self._has_qt:
+
+        if not self._has_qt and not self._has_ui:
```

Comment:
A quick comment explaining that if we don't have qt we don't even bother with dealing with a ui would be helpful (if I understood right).

Reply:
Do we even need the `and not self._has_ui` here? Maybe I'm just confusing myself, in which case a comment (as Stephane suggested) would clear it up for me.

Reply:
Same here. I fail to see the distinction between both branches. They both call `cb()`